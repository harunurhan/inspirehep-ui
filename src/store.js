import { applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';
import createHistory from 'history/createBrowserHistory';
import { routerMiddleware } from 'react-router-redux';
/* eslint-disable import/no-extraneous-dependencies */
import { createLogger } from 'redux-logger';
import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly';
/* eslint-disable import/no-extraneous-dependencies */

import reducers from './reducers';
import http from './common/http';
import queryParamsParserMiddleware from './middlewares/queryParamsParser';
import persistUserStateMiddleware, {
  reHydrateRootStateWithUser,
} from './middlewares/persistUserState';

export const thunkMiddleware = thunk.withExtraArgument(http);

export const history = createHistory();
const reduxRouterMiddleware = routerMiddleware(history);

const getMiddleware = () => {
  if (process.env.NODE_ENV === 'production') {
    return applyMiddleware(
      reduxRouterMiddleware,
      queryParamsParserMiddleware,
      persistUserStateMiddleware,
      thunkMiddleware
    );
  }
  return applyMiddleware(
    reduxRouterMiddleware,
    queryParamsParserMiddleware,
    persistUserStateMiddleware,
    thunkMiddleware,
    createLogger()
  );
};

export const store = createStore(
  reducers,
  reHydrateRootStateWithUser(),
  composeWithDevTools(getMiddleware())
);
