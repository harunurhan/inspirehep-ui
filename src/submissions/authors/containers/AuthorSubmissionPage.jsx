import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Map } from 'immutable';

import { submitAuthor } from '../../../actions/submissions';
import AuthorSubmission from '../components/AuthorSubmission';

class AuthorSubmissionPage extends Component {
  constructor(props) {
    super(props);
    this.onSubmit = this.onSubmit.bind(this);
  }

  async onSubmit(formData) {
    const { dispatch } = this.props;
    await dispatch(submitAuthor(formData));
  }

  render() {
    const { error } = this.props;
    return <AuthorSubmission error={error} onSubmit={this.onSubmit} />;
  }
}

AuthorSubmissionPage.propTypes = {
  dispatch: PropTypes.func.isRequired,
  error: PropTypes.instanceOf(Map), // eslint-disable-line react/require-default-props
};

const stateToProps = state => ({
  error: state.submissions.get('submitError'),
});

const dispatchToProps = dispatch => ({ dispatch });

export default connect(stateToProps, dispatchToProps)(AuthorSubmissionPage);
