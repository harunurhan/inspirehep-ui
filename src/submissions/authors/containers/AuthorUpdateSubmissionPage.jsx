import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Map } from 'immutable';
import { Row, Col } from 'antd';

import {
  fetchAuthorUpdateFormData,
  submitAuthorUpdate,
} from '../../../actions/submissions';
import AuthorSubmission from '../components/AuthorSubmission';

class AuthorUpdateSubmissionPage extends Component {
  static getRecordIdFromProps(props) {
    return props.match.params.id;
  }

  constructor(props) {
    super(props);
    this.onSubmit = this.onSubmit.bind(this);
  }

  componentDidMount() {
    this.dispatch(fetchAuthorUpdateFormData(this.recordId));
  }

  componentDidUpdate(prevProps) {
    const prevRecordId = AuthorUpdateSubmissionPage.getRecordIdFromProps(
      prevProps
    );
    if (this.recordId !== prevRecordId) {
      this.dispatch(fetchAuthorUpdateFormData(this.recordId));
    }
  }

  async onSubmit(formData) {
    await this.dispatch(submitAuthorUpdate(formData, this.recordId));
  }

  get dispatch() {
    const { dispatch } = this.props;
    return dispatch;
  }

  get recordId() {
    return AuthorUpdateSubmissionPage.getRecordIdFromProps(this.props);
  }

  render() {
    // TODO: display updateFormDataError ?
    const { error, updateFormData, loadingUpdateFormData } = this.props;
    return (
      !loadingUpdateFormData && (
        <Row type="flex" justify="center">
          <Col className="mt3 mb3" span={14}>
            <Row>
              <Col>
                <AuthorSubmission
                  error={error}
                  onSubmit={this.onSubmit}
                  initialFormData={updateFormData}
                />
              </Col>
            </Row>
          </Col>
        </Row>
      )
    );
  }
}

AuthorUpdateSubmissionPage.propTypes = {
  match: PropTypes.objectOf(PropTypes.any).isRequired, // eslint-disable-line react/no-unused-prop-types
  dispatch: PropTypes.func.isRequired,
  error: PropTypes.instanceOf(Map), // eslint-disable-line react/require-default-props
  updateFormData: PropTypes.instanceOf(Map), // eslint-disable-line react/require-default-props
  loadingUpdateFormData: PropTypes.bool.isRequired,
};

const stateToProps = state => ({
  error: state.submissions.get('submitError'),
  updateFormData: state.submissions.get('initialData'),
  loadingUpdateFormData: state.submissions.get('loadingInitialData'),
});

const dispatchToProps = dispatch => ({ dispatch });

export default connect(stateToProps, dispatchToProps)(
  AuthorUpdateSubmissionPage
);
