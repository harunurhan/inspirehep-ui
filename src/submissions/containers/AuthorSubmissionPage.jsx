import React, { Component } from 'react';
// import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Row, Col } from 'antd';
import { Formik } from 'formik';

import AuthorForm from '../components/AuthorForm';
import authorSchema from '../schemas/author';

class AuthorSubmissionPage extends Component {
  render() {
    return (
      <Row type="flex" justify="center">
        <Col className="mt3 mb3" span={14}>
          Author Submission
          <Formik
            initialValues={{
              display_name: 'Harun Urhan',
              field_of_research: ['hep-ph'],
            }}
            validationSchema={authorSchema}
            onSubmit={(values, actions) => {
              console.log(values);
              actions.setSubmitting(false);
            }}
            component={AuthorForm}
          />
        </Col>
      </Row>
    );
  }
}

AuthorSubmissionPage.propTypes = {
  // dispatch: PropTypes.func.isRequired,
};

const dispatchToProps = dispatch => ({ dispatch });

export default connect(null, dispatchToProps)(AuthorSubmissionPage);
