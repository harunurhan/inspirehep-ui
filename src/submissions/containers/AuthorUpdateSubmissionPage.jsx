import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Map } from 'immutable';

import {
  fetchAuthorUpdateFormData,
  submitAuthorUpdate,
} from '../../actions/submissions';
import {
  authorSubmitErrorPath,
  authorUpdateDataPath,
  loadingAuthorUpdateDataPath,
} from '../../reducers/submissions';
import AuthorSubmission from '../components/AuthorSubmission';

class AuthorUpdateSubmissionPage extends Component {
  static getRecordIdFromProps(props) {
    return props.match.params.id;
  }

  constructor(props) {
    super(props);
    this.onSubmit = this.onSubmit.bind(this);
  }

  componentDidMount() {
    const recordId = this.getRecordId();
    this.props.dispatch(fetchAuthorUpdateFormData(recordId));
  }

  componentDidUpdate(prevProps) {
    const prevRecordId = AuthorUpdateSubmissionPage.getRecordIdFromProps(
      prevProps
    );
    const recordId = this.getRecordId();
    if (recordId !== prevRecordId) {
      this.props.dispatch(fetchAuthorUpdateFormData(recordId));
    }
  }

  async onSubmit(formData) {
    // TODO: submit update!
    const recordId = this.getRecordId();
    await this.props.dispatch(submitAuthorUpdate(formData, recordId));
  }

  getRecordId() {
    return AuthorUpdateSubmissionPage.getRecordIdFromProps(this.props);
  }

  render() {
    // TODO: display updateFormDataError ?
    const { error, updateFormData, loadingUpdateFormData } = this.props;
    return (
      !loadingUpdateFormData && (
        <AuthorSubmission
          error={error}
          onSubmit={this.onSubmit}
          initialFormData={updateFormData}
        />
      )
    );
  }
}

AuthorUpdateSubmissionPage.propTypes = {
  match: PropTypes.objectOf(PropTypes.any).isRequired, // eslint-disable-line react/no-unused-prop-types
  dispatch: PropTypes.func.isRequired,
  error: PropTypes.instanceOf(Map), // eslint-disable-line react/require-default-props
  updateFormData: PropTypes.instanceOf(Map), // eslint-disable-line react/require-default-props
  loadingUpdateFormData: PropTypes.bool.isRequired,
};

const stateToProps = state => ({
  error: state.submissions.getIn(authorSubmitErrorPath),
  updateFormData: state.submissions.getIn(authorUpdateDataPath),
  loadingUpdateFormData: state.submissions.getIn(loadingAuthorUpdateDataPath),
});

const dispatchToProps = dispatch => ({ dispatch });

export default connect(stateToProps, dispatchToProps)(
  AuthorUpdateSubmissionPage
);
