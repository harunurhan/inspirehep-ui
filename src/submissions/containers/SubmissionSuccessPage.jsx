import React, { Component } from 'react';
import { Row, Col, Alert } from 'antd';

class SubmissionSuccessPage extends Component {
  render() {
    return (
      <Row type="flex" justify="center">
        <Col className="mt3 mb3" span={14}>
          <Alert
            message="Successfully submitted, thanks you for the submission!"
            type="success"
            showIcon
          />
        </Col>
      </Row>
    );
  }
}

export default SubmissionSuccessPage;
