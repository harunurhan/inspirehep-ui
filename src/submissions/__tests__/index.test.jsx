import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import { shallow, mount } from 'enzyme';
import Loadable from 'react-loadable';

import { getStore } from '../../fixtures/store';
import Submissions from '../';
import AuthorSubmissionPage from '../containers/AuthorSubmissionPage';
import SubmissionSuccessPage from '../components/SubmissionSuccessPage';
import AuthorUpdateSubmissionPage from '../containers/AuthorUpdateSubmissionPage';

// TODO: enable tests after bumping enzyme (eznyme@3.4 was not stable at the time of the TODO)
describe('Submissions', () => {
  it('renders initial state', () => {
    const component = shallow(<Submissions />);
    expect(component).toMatchSnapshot();
  });

  xit('navigates to AuthorSubmissionPage when /submissions/author', async done => {
    const wrapper = mount(
      <Provider store={getStore()}>
        <MemoryRouter initialEntries={['/submissions/author']} initialIndex={0}>
          <Submissions />
        </MemoryRouter>
      </Provider>
    );
    await Loadable.preloadAll();
    wrapper.update();

    expect(wrapper.find(AuthorSubmissionPage)).toExist();

    done();
  });

  xit('navigates to AuthorUpdateSubmissionPage when /submissions/author/:id', async done => {
    const wrapper = mount(
      <Provider store={getStore()}>
        <MemoryRouter
          initialEntries={['/submissions/author/1']}
          initialIndex={0}
        >
          <Submissions />
        </MemoryRouter>
      </Provider>
    );
    await Loadable.preloadAll();
    wrapper.update();

    expect(wrapper.find(AuthorUpdateSubmissionPage)).toExist();

    done();
  });

  xit('navigates to SubmissionSuccessPage when /submissions/success', async done => {
    const wrapper = mount(
      <Provider store={getStore()}>
        <MemoryRouter
          initialEntries={['/submissions/success']}
          initialIndex={0}
        >
          <Submissions />
        </MemoryRouter>
      </Provider>
    );
    await Loadable.preloadAll();
    wrapper.update();

    expect(wrapper.find(SubmissionSuccessPage)).toExist();

    done();
  });
});
