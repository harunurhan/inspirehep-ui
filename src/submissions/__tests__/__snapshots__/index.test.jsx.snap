// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Submissions renders initial state 1`] = `
<div
  className="w-100"
>
  <Route
    component={[Function]}
    exact={true}
    path="/submissions/authors"
  />
  <Route
    component={[Function]}
    exact={true}
    path="/submissions/authors/:id"
  />
  <Route
    component={[Function]}
    exact={true}
    path="/submissions/success"
  />
</div>
`;
